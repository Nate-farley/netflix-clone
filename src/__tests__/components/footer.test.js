import React from "react";
import { render } from "@testing-library/react";
import { Footer } from '../../components';

describe('<Footer />', ()=> {
    it('renders the <Footer /> with populated data', () => {
        const { container, getByText } = render( 
        <Footer>
            <Footer.Title>Questions? Contact Us.</Footer.Title>
            <Footer.Break />
            <Footer.Row>
                <Footer.Column>
                    <Footer.Link herf="#">FAQ</Footer.Link>
                    <Footer.Link herf="#">Investor Relations</Footer.Link>
                    <Footer.Link herf="#">Ways to Watch</Footer.Link>
                    <Footer.Link herf="#">Coporate Information</Footer.Link>
                    <Footer.Link herf="#">Netflix Originals</Footer.Link>
                </Footer.Column>

                <Footer.Column>
                    <Footer.Link herf="#">Help Center</Footer.Link>
                    <Footer.Link herf="#">Jobs</Footer.Link>
                    <Footer.Link herf="#">Terms of Use</Footer.Link>
                    <Footer.Link herf="#">Contact Us</Footer.Link>
                </Footer.Column>

                <Footer.Column>
                    <Footer.Link herf="#">Account</Footer.Link>
                    <Footer.Link herf="#">Redeem Gift Cards</Footer.Link>
                    <Footer.Link herf="#">Privacy</Footer.Link>
                    <Footer.Link herf="#">Speed Test</Footer.Link>
                </Footer.Column>

                <Footer.Column>
                    <Footer.Link herf="#">Meida Center</Footer.Link>
                    <Footer.Link herf="#">Buy Gift Cards</Footer.Link>
                    <Footer.Link herf="#">Cookie Preferences</Footer.Link>
                    <Footer.Link herf="#">Legal Notices</Footer.Link>
                </Footer.Column>
            </Footer.Row>
            <Footer.Break />
            <Footer.Text>Netflix United Kingdom</Footer.Text>
        </Footer>
        );
        
        expect(getByText('Questions? Contact Us.')).toBeTruthy();
        expect(getByText('FAQ')).toBeTruthy();
        expect(getByText('Investor Relations')).toBeTruthy();
        expect(getByText('Ways to Watch')).toBeTruthy();
        expect(getByText('Coporate Information')).toBeTruthy();
        expect(getByText('Netflix Originals')).toBeTruthy();
        expect(container.firstChild).toMatchSnapshot();

    })
})