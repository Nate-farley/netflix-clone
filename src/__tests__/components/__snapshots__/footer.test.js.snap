// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<div
  class="footer__Container-sc-eh6b4z-0 bjISUr"
>
  <p
    class="footer__Title-sc-eh6b4z-4 dLPDHa"
  >
    Questions? Contact Us.
  </p>
  <div
    class="footer__Break-sc-eh6b4z-6 jiQmMZ"
  />
  <div
    class="footer__Row-sc-eh6b4z-2 jDlsgE"
  >
    <div
      class="footer__Column-sc-eh6b4z-1 dIxCTP"
    >
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        FAQ
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Coporate Information
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Netflix Originals
      </a>
    </div>
    <div
      class="footer__Column-sc-eh6b4z-1 dIxCTP"
    >
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Help Center
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Jobs
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Contact Us
      </a>
    </div>
    <div
      class="footer__Column-sc-eh6b4z-1 dIxCTP"
    >
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Account
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Privacy
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Speed Test
      </a>
    </div>
    <div
      class="footer__Column-sc-eh6b4z-1 dIxCTP"
    >
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Meida Center
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-sc-eh6b4z-3 iqjmuP"
      >
        Legal Notices
      </a>
    </div>
  </div>
  <div
    class="footer__Break-sc-eh6b4z-6 jiQmMZ"
  />
  <p
    class="footer__Text-sc-eh6b4z-5 dsycrP"
  >
    Netflix United Kingdom
  </p>
</div>
`;
